### PR TITLE
feat: add support for locally hosted point cloud models to Cognite3dViewer

### PR DESCRIPTION
**Pull request by larsmoa**
- Also adds unit test for point clouds using Cognite3dViewer
- Bump puppeteer

### CODE REVIEW

File: examples/src/utils/isUrlPointCloudModel.ts [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+ * Returns true if the URL provided points to a point cloud model.
+ * @param addModelOptions 
+ * @returns 
+ */
+export async function isUrlPointCloudModel(modelBaseUrl: string) {
```
Comment (haakonflatval-cognite) [inline examples/src/utils/isUrlPointCloudModel.ts:6]: Could we name this `isLocalUrlPointCloudModel`? Just to ensure no one tries using it for remote requests that e.g. may require authentication.

File: examples/src/utils/ModelUi.ts [hunk 2]
```diff
@@ -136,6 +137,11 @@ export class ModelUi {
 
 }
 
+async function loadLocalModel(viewer: Cognite3DViewer, addModelOptions: AddModelOptions): Promise<CognitePointCloudModel | Cognite3DModel> {
```
Comment (haakonflatval-cognite) [inline examples/src/utils/ModelUi.ts:140]:
```suggestion
async function addLocalModel(viewer: Cognite3DViewer, addModelOptions: AddModelOptions): Promise<CognitePointCloudModel | Cognite3DModel> {
```

File: examples/src/utils/ModelUi.ts [hunk 3]
```diff
@@ -114,7 +115,7 @@ export class ModelUi {
 
   async addModel(options: AddModelOptions) {
     try {
-      const model = options.localPath !== undefined ? await this._viewer.addCadModel(options) : await this._viewer.addModel(options);
+      const model = options.localPath !== undefined ? await loadLocalModel(this._viewer, options) : await this._viewer.addModel(options);
```
Comment (haakonflatval-cognite) [inline examples/src/utils/ModelUi.ts:118]:
```suggestion
      const model = options.localPath !== undefined ? await addLocalModel(this._viewer, options) : await this._viewer.addModel(options);
```

**Review by haakonflatval-cognite — APPROVED** [review]
A couple of comments/suggestions. Otherwise good

File: examples/src/pages/e2e/Cognite3DTestViewer.tsx [hunk 4]
```diff
@@ -60,30 +64,45 @@ export function Cognite3DTestViewer(props: Props) {
...
 
     return () => {
       viewer && viewer.dispose();
     };
-  }, [geometryFilter, initializeCallback, modelAddedCallback, modelUrls, viewerOptions, fitCameraToModel]);
+  }, [geometryFilter, initializeCallback, cadModelAddedCallback, pointCloudModelAddedCallback, modelUrls, viewerOptions, fitCameraToModel]);
 
   const readyForScreenshot = loadingState.itemsLoaded > 0 &&
     loadingState.itemsLoaded === loadingState.itemsRequested;
+  console.log(JSON.stringify(loadingState), readyForScreenshot);
```
Comment (christjt) [inline examples/src/pages/e2e/Cognite3DTestViewer.tsx:105]: Should this be here?

**Review by christjt — APPROVED** [review]

**Pull request merged**